PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheck.java

```diff
@@ -0,0 +1,120 @@
+            } catch (JsonParseException e) {
+                LOGGER.log(Level.SEVERE, "Configuration file is not a valid JSON at path: " + LoadJsonConfigFile.CONFIG_FILE_PATH
+                        + ". Please check the file format. Error: " + e.getMessage(), e);
+            } catch (IOException e) {
+                LOGGER.log(Level.SEVERE, "IO error while reading configuration file from path: " + LoadJsonConfigFile.CONFIG_FILE_PATH
+                        + ". Please check file permissions and retry. Error: " + e.getMessage(), e);
+            } catch (Exception e) {
+                LOGGER.log(Level.SEVERE, "Unexpected error while loading configuration for rule '" + ruleName
+                        + "'. Please investigate further. Error: " + e.getMessage(), e);
+            }
+        }
+
+        /**
+         * Constructor to initialize the visitor with the holder and isOnTheFly flag.
+         * @param holder     ProblemsHolder to register problems
+         * @param isOnTheFly boolean to check if the inspection is on the fly
+         */
+        public ConnectionStringVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+            this.isOnTheFly = isOnTheFly;
+        }
+
+        /**
+         * This method visits the element in the code.
+         * It checks if the method call is a connection string call and if the class is an Azure client.
+         * If both conditions are met, a problem is registered with the suggestion message.
+         * @param element PsiElement to visit
+         */
+        @Override
+        public void visitElement(@NotNull PsiElement element) {
+            super.visitElement(element);
+
+            // check if an element is an azure client
+            if (element instanceof PsiMethodCallExpression) {
+
+                PsiMethodCallExpression methodCallExpression = (PsiMethodCallExpression) element;
+                PsiReferenceExpression methodExpression = methodCallExpression.getMethodExpression();
+
+                // resolvedMethod is the method that is being called
+                PsiElement resolvedMethod = methodExpression.resolve();
+
+                // check if the method is a connectionString call
+                if (resolvedMethod != null && resolvedMethod instanceof PsiMethod && ((PsiMethod) resolvedMethod).getName().equals(METHOD_TO_CHECK)) {
+                    PsiMethod method = (PsiMethod) resolvedMethod;
+
+                    // containingClass is the client class that is being called. check if the class is an azure client
+                    PsiClass containingClass = method.getContainingClass();
+
+                    // compare the package name of the containing class to the azure package name from the configuration file
+                    if (containingClass != null && containingClass.getQualifiedName() != null && containingClass.getQualifiedName().startsWith(LoadJsonConfigFile.PACKAGE_NAME)) {
```

Comment:
Why are we checking the containing class with package name.
Wouldn't the class name be something like `BlobClient` and package name is `com.azure`

Reply:
ContainingClass.getQualifiedName results in eg  "com.azure..storage....CLientName" that's why I check if the containignClass.qualifiedName startswith "com.azure"